src/OpenIddict.Client/OpenIddictClientBuilder.cs

```diff
@@ -192,14 +192,18 @@ public OpenIddictClientBuilder AddDevelopmentEncryptionCertificate()
     /// Registers (and generates if necessary) a user-specific development encryption certificate.
     /// </summary>
     /// <param name="subject">The subject name associated with the certificate.</param>
+    /// <param name="notBefore">The NotBefore of the certificate.</param>
     /// <returns>The <see cref="OpenIddictClientBuilder"/> instance.</returns>
-    public OpenIddictClientBuilder AddDevelopmentEncryptionCertificate(X500DistinguishedName subject)
+    public OpenIddictClientBuilder AddDevelopmentEncryptionCertificate(X500DistinguishedName subject, DateTimeOffset? notBefore = null)
```

Comment:
Can it be an overload?

```cs
public OpenIddictClientBuilder AddDevelopmentEncryptionCertificate(
    X500DistinguishedName subject
) => AddDevelopmentEncryptionCertificate(subject, DateTimeOffset.UtcNow);

public OpenIddictClientBuilder AddDevelopmentEncryptionCertificate(X500DistinguishedName subject, DateTimeOffset notBefore)
{
...
}
```

Reply:
Could be, but it honestly looks very weird.

Another approach: move the entire method body inside the `return Configure(options => ...)` delegate. From there, you'd be able to access the `TimeProvider` via `options.TimeProvider`, tho' that only works if the provider was set before the inline `IOptionsConfigure<T>` registered by this extension kicks in. Could be a reasonable compromise.

Reply:
I moved it inside the Configure method, as you suggested. The issue is that we resolve `TimeProvider` in the `IPostConfigureOptions`, therefore it will never be populated before this `Configure` is executed.

To handle this I used this API:
```cs
Services.AddOptions<OpenIddictClientOptions>().Configure<IServiceProvider>((options, serviceProvider) => {
...
});
```

I used the `DateTimeOffsetUtcNow.LocalDateTime`. It should have a correct `Kind`:
https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUH1TATgAoAVASwFsBTABTAHsA3CuGsAOkM4HEbgAGUYBjAIYAbAHKMA7iQCUnYeIkARMcBqVanANIUocBQG4gA==

Reply:
Great, thanks! It's a very reasonable approach.